mesop/labs/text_to_image.py

```diff
@@ -0,0 +1,132 @@
+import types
+from typing import Callable, Generator, cast
+
+import mesop as me
+
+
+@me.stateclass
+class State:
+  input: str
+  output: str
+  textarea_key: int
+
+
+def text_to_image(
+  transform: Callable[[str], Generator[str, None, None] | str],
+  *,
+  title: str | None = None,
+):
+  """Creates a simple UI which takes in a text input and returns an image output.
+
+  This function creates event handlers for text input and output operations
+  using the provided function `transform` to process the input and generate the image
+  output.
+
+  Args:
+    transform: Function that takes in a string input and returns a URL to an image
+    or a base64 encoded image.
+    title: Headline text to display at the top of the UI.
+  """
+
+  def on_input(e: me.InputEvent):
+    state = me.state(State)
+    state.input = e.value
+
+  def on_click_generate(e: me.ClickEvent):
+    state = me.state(State)
+    output = transform(state.input)
+    if isinstance(output, types.GeneratorType):
+      for val in output:
+        state.output = val
+        yield
+    else:
+      # `output` is a str, however type inference doesn't
+      # work w/ generator's unusual ininstance check.
+      state.output = cast(str, output)
+      yield
+
+  def on_click_clear(e: me.ClickEvent):
+    state = me.state(State)
+    state.input = ""
+    state.output = ""
+    state.textarea_key += 1
+
+  with me.box(
```

Comment:
I'm wondering if we should reuse some of these styles across the i/o components. I think if we do a third i/o component, it may make sense to dedupe some of the styles (where applicable) to make it easier to maintain.

Reply:
I agree that we can leave as is for now. And agree that we can dedupe some stuff that is reusable.